extension/page-detect.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+window.pageDetect = (() => {
+	const isGistCheck = location.hostname === 'gist.github.com';
+
+	const isGist = () => isGistCheck;
+
+	const isDashboard = () => location.pathname === '/' || /^(\/orgs\/[^\/]+)?\/dashboard/.test(location.pathname);
+
+	const isRepo = () => !isGist() && /^\/[^/]+\/[^/]+/.test(location.pathname);
```

Comment:
@sindresorhus With this pseudo code, it's not going to work, because when `isIssue()` function is called it should take the latest `location.pathname` and the `repoPath` won't be updated. What we could do instead is expose a factory-like function which we'd call on every GitHub injection and it will return the actual object with the checking functions which will hold the latest repo path in a closure. This would be quite optimal as it make as few checks as necessary and some checks could be made only once. However I'm not sure if it's worth the effort. The other option is to extract `repoPath` every time.